utils/components.py

```diff
@@ -394,7 +394,7 @@ def test_sdg_params(sdg_batch_size: int, sdg_num_workers: int):
     if (
         sdg_batch_size < 1
         or sdg_batch_size > 4096
-        or sdg_num_workers < 2
+        or sdg_num_workers < 1
```

Comment:
Unfortunately, we can't allow this because this causes InstructLab to disable batching and then sends each completion request in a separate HTTP call.

Reply:
Oh I see. This is intentional based on the docstring. What's the usecase for disabling batching?

Reply:
The idea is to reduce the load on the teacher model to avoid SDG failing due to timeout errors (see https://issues.redhat.com/browse/RHOAIENG-20555#)

Also, the error message in test_sdg_params says `sdg_batch_size must be a value between 1-4096`

Reply:
We were told by RHEL AI that we always want batching enabled which is why I added this check originally since they said an HTTP request per completion creates a high load on the inference endpoint.

Reply:
I've modified the code to ensure that sdg_num_workers is a value between 2-10